src/portalbackend/PortalBackend.DBAccess/Repositories/OfferSubscriptionsRepository.cs

```diff
@@ -105,15 +105,19 @@ sorting switch
             .SingleOrDefaultAsync();
 
     /// <inheritdoc />
-    public Task<(OfferSubscriptionStatusId OfferSubscriptionStatusId, bool IsSubscribingCompany, bool IsValidSubscriptionId)> GetCompanyAssignedAppDataForCompanyUserAsync(Guid subscriptionId, Guid userCompanyId) =>
+    public Task<(OfferSubscriptionStatusId OfferSubscriptionStatusId, bool IsSubscribingCompany, bool IsValidSubscriptionId, Guid ConnectorId, Guid IdentityId)> GetCompanyAssignedOfferSubscriptionDataForCompanyUserAsync(Guid subscriptionId, Guid userCompanyId) =>
         _context.OfferSubscriptions
             .Where(os =>
                 os.Id == subscriptionId
             )
-            .Select(os => new ValueTuple<OfferSubscriptionStatusId, bool, bool>(
+            .Select(os => new ValueTuple<OfferSubscriptionStatusId, bool, bool, Guid, Guid>(
                 os.OfferSubscriptionStatusId,
                 os.CompanyId == userCompanyId,
-                true
+                true,
+                os.ConnectorAssignedOfferSubscriptions.Where(caos => caos.Connector!.StatusId != ConnectorStatusId.INACTIVE).Select(caos =>
+                    caos.Connector!.Id).SingleOrDefault(),
```

Comment:
the datamodel allows the assignment of multiple connectors to one offer subscription, if we do a singleOrDefault and the database contains multiple entries we'll throw a 500 error, better:

```suggestion
                os.ConnectorAssignedOfferSubscriptions.Where(caos => caos.Connector!.StatusId != ConnectorStatusId.INACTIVE).Select(caos =>
                    caos.Connector!.Id),
```